apps/site/assets/ts/stop/components/DisplayTime.tsx

```diff
@@ -0,0 +1,162 @@
+/**
+ * <DisplayTime /> Renders a UI element composing several different aspects of a
+ * "departure" time. The following diagrams depict the layout used for the list
+ * on the realtime tracking page, but can be further adjusted with CSS.
+ *
+  |------------|---------------------------------------------------------|
+  | maybe Icon | {maybe "Delayed"} Time1 maybe Time2                     |
+  |------------|---------------------------------------------------------|
+  |            | {maybe "Delayed"} {maybe Delayed Details} maybe Details |
+  |------------|---------------------------------------------------------|
+
+  Example uses:
+  |------|-----------|    |------|---------|
+  |      | 12:59 AM  |    | Icon | 20 min  |
+  |------|-----------|    |------|---------|
+  |      | Tomorrow  |
+  |------|-----------|
+
+  |------|------------------- |    |------|-----------------------------|
+  | Icon | Delayed 4:24PM     |    | Icon | 28 min                      |
+  |------|--------------------|    |------|-----------------------------|
+  |      | ~~4:15PM~~ Track 3 |    |      | Delayed 11:14AM ~~11:05AM~~ |
+  |------|--------------------|    |------|-----------------------------|
+ */
+
+import React, { ReactElement, createContext, useContext } from "react";
+import { differenceInSeconds, isTomorrow, secondsInHour } from "date-fns";
+import { DepartureInfo } from "../../models/departureInfo";
+import realtimeIcon from "../../../static/images/icon-realtime-tracking.svg";
+import SVGIcon from "../../helpers/render-svg";
+import {
+  departureInfoToTime,
+  displayInfoContainsPrediction
+} from "../../helpers/departureInfo";
+import BasicTime from "./BasicTime";
+
+interface DisplayTimeProps {
+  departure: DepartureInfo;
+  isCR: boolean;
+  targetDate?: Date | undefined;
+}
+export interface DepartureContextProps extends DisplayTimeProps {
+  time: Date | undefined;
+  isLessThanHourAway: boolean | undefined;
+}
+
+/**
+ * This context is meant to capture some of the values that affect the UI design
+ * so we populate it with a few extra values to avoid recalculation
+ */
+export const DepartureContext = createContext<DepartureContextProps>({
+  departure: {},
+  time: undefined,
+  isCR: false,
+  isLessThanHourAway: false,
+  targetDate: new Date()
+});
+
+function TimeCountdown(): JSX.Element {
+  const { time, isCR, targetDate } = useContext(DepartureContext);
+  return (
+    <BasicTime
+      displayType={isCR ? "absolute" : "relative"}
```

Comment:
What is the purpose of relative?  We don't display timezones anywhere and I wonder if this could lead to confusion.  Also talking to Anna in the past we have favored displaying time in the local timezone of the MBTA.

Reply:
relative is stuff like `3 min`, `Arriving`, etc... as opposed to `4:23 PM`.

Reply:
Okay, I mis-understood what this purpose was.